src/main/java/com/adobe/cq/testing/client/security/AbstractAuthorizable.java

```diff
@@ -236,17 +236,20 @@ private String getAuthorizableType(String authorizableId) throws ClientException
     }
 
     private JsonNode getAutorizablesWithRetry(final String query) throws ClientException, InterruptedException {
+        final JsonNode[] jsonNode = new JsonNode[1];
```

Comment:
Not sure I understand the role of the array. FWIU, you want to save the result of `getAuthorizables()` and return it in the end. Wouldn't a simple reference be enough for that?

Reply:
@volteanu  element accessed in inner class need to be final or effectively final, thus the only way is to store it in an array object that is final and then read it after that.